### PR TITLE
[BD-46] feat: add Layout component

### PR DESCRIPTION
**Pull request by monteri**
## Description

Add `Layout` component that is wrapper for `Row` and `Col` from `react-bootstrap`. It provides interface to pass all breakpoints data to the `Layout` instead of to each component separately.

### Deploy Preview

https://deploy-preview-1659--paragon-openedx.netlify.app/components/layout/

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @monteri!


When this pull request is ready, tag your edX technical lead.

File: src/Layout/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Basic usage
+
+```jsx live
+  <Layout
+    lg={{ span: [4, 4, 4], offset: [0, 0, 0] }}
+    md={{ span: ['auto', 'auto', 'auto'], offset: [0, 0, 0] }}
+    sm={{ span: [8, 4, 6], offset: [0, 0, 6] }}
+    xs={{ span: [4, 4, 4], offset: [0, 0, 0] }}
+    xl={{ span: [3, 6, 3] }}
```
Comment (adamstankiewicz) [inline src/Layout/README.md:23]: Based on PWG meeting, let's move the API to support something like:

```jsx
<Layout
    lg={[{ span: 4, offset: 0}, { span: 4, offset: 0 }, { span: 4: offset: 0 }]}
/>
```

to make it more clear how the `span`/`offset` relate to each other per column. It's also a bit more in line with the `react-bootstrap` API on the `Col` component.

**Review by adamstankiewicz — APPROVED** [review]
LGTM! We'll can update the documentation to describe the concept of `Row` and `Col` as a separate chunk of work / issue.

**Comment by openedx-webhooks** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1659&entry.1484655318=https://github.com/monteri&entry.752974735=2022-10-03+13:37&entry.1917517419=2022-10-21+12:35&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**